outbound/hmail.js

```diff
@@ -287,19 +287,14 @@ class HMailItem extends events.EventEmitter {
 
         this.force_tls = this.get_force_tls(mx)
 
-        // Allow transaction notes to set outbound IP
-        if (!mx.bind && this.todo.notes.outbound_ip) {
-            mx.bind = this.todo.notes.outbound_ip;
+        if (this.todo.notes.outbound_ip) {
+            this.logerror(`notes.outbound_ip is deprecated. Use get_mx.bind instead!`);
+            if (!mx.bind) mx.bind = this.todo.notes.outbound_ip;
         }
 
         // Allow transaction notes to set outbound IP helo
-        if (!mx.bind_helo){
-            if (this.todo.notes.outbound_helo) {
-                mx.bind_helo = this.todo.notes.outbound_helo;
-            }
-            else {
-                mx.bind_helo = net_utils.get_primary_host_name();
-            }
+        if (this.todo.notes.outbound_helo) {
+            mx.bind_helo = this.todo.notes.outbound_helo;
```

Comment:
setting it unconditionally here makes the docs wrong (they say `get_mx` has precedence).

Reply:
> The difficult question is how and why others use outbound ip, if at all

Exactly. That's why I want to put that error log in there. If someone is using it, they may/should notice the error messages and then raise an objection. Then maybe we'll also gain understanding of the use case and hopefully the reasoning behind the "don't set this manually" comment in the docs.

Reply:
makes sense!